distributed/comm/ucx.py

```diff
@@ -158,13 +158,9 @@ def peer_address(self) -> str:
                 )
 
                 # Send meta data
+                cuda_frames = [hasattr(f, "__cuda_array_interface__") for f in frames]
```

Comment:
Would actually just go ahead and make this an array since we need that later anyways.

```suggestion
                cuda_frames = np.array([hasattr(f, "__cuda_array_interface__") for f in frames], dtype=bool)
```